### PR TITLE
Support all float dtypes in `F.mean_squared_error`

### PR DESCRIPTION
**Pull request by takagi**
This PR is a part of #4582 and generalizes `F.mean_squared_error` to support all float dtypes other than `float32`, with loosing the condition of inputs.

### CODE REVIEW

File: tests/chainer_tests/functions_tests/loss_tests/test_mean_squared_error.py [hunk 1]
```diff
@@ -46,9 +58,11 @@ def test_forward_gpu(self):
         self.check_forward(cuda.to_gpu(self.x0), cuda.to_gpu(self.x1))
 
     def check_backward(self, x0_data, x1_data):
+        atol = self.backward_tols['atol']
+        rtol = self.backward_tols['rtol']
         gradient_check.check_backward(
             functions.mean_squared_error,
-            (x0_data, x1_data), None, eps=1e-2)
+            (x0_data, x1_data), None, eps=1e-2, atol=atol, rtol=rtol)
```
Comment (toslunar) [inline tests/chainer_tests/functions_tests/loss_tests/test_mean_squared_error.py]: Why don't you use `**self.backward_tols`?

Comment (takagi) [reply tests/chainer_tests/functions_tests/loss_tests/test_mean_squared_error.py]: Yes, it's enough!

**Comment by okuta** [comment]
jenkins, test this please.

**Comment by okuta** [comment]
LGTM!

**Comment by takagi** [comment]
Would you merge this?

**Pull request merged**